pgmpy/readwrite/ProbModelXML.py

```diff
@@ -383,8 +387,110 @@ def _add_decision_criteria(self, criteria_dict):
             criteria_tag = etree.SubElement(decision_tag, 'Criterion', attrib={'name': criteria})
             self._add_additional_properties(criteria_tag, criteria_dict[criteria])
 
-    def _add_potential(self):
-        pass
+    def _add_potential(self, potential, parent_tag):
+        """
+        Adds Potentials to the ProbModelXML.
+
+        Parameters
+        ----------
+        potential: dict
+            Dictionary containing Potential data.
+            For example: {'role': 'Utility',
+                          'Variables': ['D0', 'D1', 'C0', 'C1'],
+                          'type': 'Tree/ADD',
+                          'UtilityVaribale': 'U1'}
+        parent_tag: etree Element
+            etree element which would contain potential tag
+            For example: <Element Potentials at 0x7f315fc44b08>
+                         <Element Branch at 0x7f315fc44c88>
+                         <Element Branch at 0x7f315fc44d88>
+                         <Element Subpotentials at 0x7f315fc44e48>
+
+        Examples
+        -------
+        >>> writer = ProbModelXMLWriter(model)
+        >>> writer._add_potential(potential, parent_tag)
+        """
+        potential_type = potential['type']
+        try:
+            potential_tag = etree.SubElement(parent_tag, 'Potential', attrib={
+                'type': potential['type'], 'role': potential['role']})
+        except KeyError:
+            potential_tag = etree.SubElement(parent_tag, 'Potential', attrib={
+                'type': potential['type']})
+        if 'Comment' in potential:
+            etree.SubElement(potential_tag, 'Comment').text = potential['Comment']
+        if 'AdditionalProperties' in potential:
+            self._add_additional_properties(potential_tag, potential['AdditionalProperties'])
+        if potential_type == "delta":
+            etree.SubElement(potential_tag, 'Variable', attrib={'name': potential['Variable']})
+            if 'State' in potential:
```

Comment:
The code snippet

```python
if 'Someone' in potential:
    # then do something
```

is repeated multiple times over here for ```State```, ```StateIndex```, ```NumericValue``` and lots of other places. Avoid repeating code as much as possible. One possible solution, would be to have a nested function such as 

```python
def _add_element(potential, var, potential_tag):
    if var  in potential:
        etree.Subelement(potential_tag, var).text = potential[var]
```

and call method ```_add_element``` as required such as ```_add_element(potential, 'State', potential_tag)```